packages/react/src/__tests__/ReactProfiler-test.internal.js

```diff
@@ -2482,9 +2482,7 @@ describe('Profiler', () => {
         throwInOnWorkStarted = true;
         expect(Scheduler).toFlushAndThrow('Expected error onWorkStarted');
         // Rendering was interrupted by the error that was thrown
-        expect(Scheduler).toHaveYielded([]);
```

Comment:
The old behavior was because the task that re-throws the error has ImmediatePriority, which would force a yield.